src/eva/transforms/select_time.py

```diff
@@ -0,0 +1,87 @@
+# (C) Copyright 2021-2022 NOAA/NWS/EMC
+#
+# (C) Copyright 2021-2022 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration. All Rights Reserved.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+from eva.transforms.transform_utils import parse_for_dict, split_collectiongroupvariable
+from eva.transforms.transform_utils import replace_cgv
+from eva.utilities.config import get
+from eva.utilities.logger import Logger
+
+
+# --------------------------------------------------------------------------------------------------
+#  Select a variable by the Time dimension in form YYYYMMDDHH.  If the input is a single time
+#  then values for the specified variable will be selected for that cycle and added to
+#  data_collections.
+#
+#  If two cycle times are included they will be used as the bounds for a time slice, and the
+#  average of that span will be added to the data_collections.  Note that when specifying a slice
+#  the order must be older (lower) to newer (higher).
+
+def select_time(config, data_collections):
+
+    # Create a logger
+    logger = Logger('SelectTimeTransform')
+
+    # Parse config for dictionary
+    [collections, groups, variables] = parse_for_dict(config, logger)
+
+    # Parse config for the expression and new collection/group/variable naming
+    new_name_template = get(config, logger, 'new name')
+    starting_field_template = get(config, logger, 'starting field')
+
+    # Get cycles from yaml file and confirm there are 1 or 2 times
+    cyc = get(config, logger, 'cycles')
```

Comment:
That would be fine.  So the yaml parsing would try `cycle` first, and if not found would try `start cycle` and `end cycle`, right?  

I'm not sure `window length` would work in all cases -- occasionally there are missing cycles in the data and `cycle interval` would have to be specified as well.  My thinking so far is that the supporting Mon 2.0 infrastructure would determine which cycles are available and generate the corresponding yaml files for image generation.

Reply:
I don't see a way in `get` (in `config.py`) to continue if a key isn't found rather than `sys.exit`.  Unless I missed an established solution I'm going to modify `get` to add an `abort_on_failure` parameter (default True).

Reply:
I think you can do:
```
cyc = get(config, logger, 'cycle')
if not cyc:
    start_cyc = get(config. logger, 'start cycle')
...

if not cyc and not start_cyc and not end_cyc:
   sys.exit()
```
(pseudo code but do you get what I"m saying?)

Reply:
perhaps I'm misreading your comment (sorry), can you add an optional extra argument that is the default which is empty or something like that?

Reply:
I think we're on the same page.  The problem is if `get `in this line
`cyc = get(config, logger, 'cycle')` 
doesn't find `cycle`, `get `fires `logger.abort`.  I'm going to add a parameter to `get` to call/not call `logger.abort` with the default set to the current behavior (call `logger.abort`).

Reply:
Changes have been made.    Please see what you think.